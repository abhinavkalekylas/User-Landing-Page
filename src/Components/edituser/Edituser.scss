$text-color: black; 

.edituser_container {
    margin-top: 1rem;

    .form, .edituser_label {
        color: $text-color;
    }

    .input {
        display: flex;
        flex-direction: row;
    }

    .edituser_label {
        font-size: 1.2rem;
        display: flex;
        justify-content: center;
        align-items: center;
        margin: 0.5rem;
    }

    .btn_edit {
        height: 3rem;
        width: 8rem;
        margin: 1rem 10rem;   
    }
}
