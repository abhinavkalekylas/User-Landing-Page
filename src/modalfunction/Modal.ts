export interface User {
  id: number;
  name: string;
  email: string;
  gender: GenderType;
  status: StatusType;
}

export enum GenderType {
  MALE = "male",
  FEMALE = "female",
}

export enum StatusType {
  ACTIVE = "active",
  INACTIVE = "inactive",
}

export const validateEmail = (email: string) => {
  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/i;
  if (regex.test(email)) {
    return true;
  }
  return false;
};
