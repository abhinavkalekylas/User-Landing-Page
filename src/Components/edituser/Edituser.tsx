import React from "react";
import { User } from "../../modalfunction/Modal";
import ModalTemplate from "../modaltemplate/ModalTemplate";
import EdituserForm from "./edituserform/EdituserForm";

const EditUser = ({
  userData,
  editUserModal,
  closeModal,
}: {
  userData: User;
  editUserModal: boolean;
  closeModal: () => void;
}) => {
  return (
    <ModalTemplate
      title_name={"Edit User"}
      userModal={editUserModal}
      closeModal={closeModal}
    >
      <EdituserForm userData={userData} closeModal={closeModal} />
    </ModalTemplate>
  );
};

export default EditUser;
