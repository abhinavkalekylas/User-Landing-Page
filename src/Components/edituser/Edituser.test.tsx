import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter as Router } from "react-router-dom";
import { GenderType, StatusType, User } from "../../modalfunction/Modal";
import Edituser from "./Edituser";

const userData: User = {
  id: 0,
  name: "shivam sharma",
  email: "shivam@gmail.com",
  gender: GenderType.MALE,
  status: StatusType.ACTIVE,
};

let wrapper: any;
const props = {
  userData,
  editUserModal: true,
  closeModal: jest.fn(),
};
beforeEach(() => {
  wrapper = shallow(
    <Router>
      <Edituser {...props} />
    </Router>
  );
});

// snapshot testing
it("renders correctly Edituser component", () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});
