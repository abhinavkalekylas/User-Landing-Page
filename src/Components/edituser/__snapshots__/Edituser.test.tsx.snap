// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Edituser component 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <EditUser
    closeModal={[MockFunction]}
    editUserModal={true}
    userData={
      Object {
        "email": "shivam@gmail.com",
        "gender": "male",
        "id": 0,
        "name": "shivam sharma",
        "status": "active",
      }
    }
  />
</Router>
`;
